renovate fails to update yarn.lock when using yarn-path

**What Renovate type are you using?**
Renovate CLI

**Describe the bug**
We have a `.yarnrc` file checked-in in our repository, which sets `yarn-path` to a single yarn bundle which is also checked-in in the repository (i.e. `./yarn.js`).
When renovate tries to update the lock file, the following error will be logged:

```
INFO: lock file error (repository=our-org/our-repo, dependencies=@types/react, branch=renovate/react-16.x)
       "cmd": "node /project/node_modules/yarn/bin/yarn.js install --ignore-scripts --ignore-engines --ignore-platform --mutex file:/tmp/.yarn-mutex",
       "err": {
         "name": "ChildProcessError",
         "code": 1,
         "childProcess": {},
         "stdout": "",
         "stderr": "internal/modules/cjs/loader.js:596\n    throw err;\n    ^\n\nError: Cannot find module '/tmp/tmp-18hUubtx7hn71/yarn.js'\n    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:594:15)\n    at Function.Module._load (internal/modules/cjs/loader.js:520:25)\n    at Function.Module.runMain (internal/modules/cjs/loader.js:744:10)\n    at startup (internal/bootstrap/node.js:238:19)\n    at bootstrapNodeJSCore (internal/bootstrap/node.js:572:3)\n"
       },
       "stdout": undefined,
       "stderr": undefined,
       "type": "yarn"
```

We tried to set the `yarnrc` option in `renovate.json` but it seems like the `.yarnrc` file found in the repository has higher precedence then this option.

------------

As renovate comes with its own version of yarn I guess it would be save when renovate strips the `yarn-path` option from the `.yarnrc` file and uses its own yarn version. I guess this should solve the issue.

--- a/lib/manager/npm/post-update/index.js
+++ b/lib/manager/npm/post-update/index.js
@@ -176,7 +176,9 @@ async function writeExistingFiles(config, packageFiles) {
       logger.debug(`Writing .yarnrc to ${basedir}`);
       await fs.outputFile(
         upath.join(basedir, '.yarnrc'),
-        packageFile.yarnrc.replace('--install.pure-lockfile true', '')
+        packageFile.yarnrc
+          .replace('--install.pure-lockfile true', '')
+          .replace(/^yarn-path.*$/m, '')
       );
     }
     const { npmLock } = packageFile;

